libs/ha/ha_common_libs_k8s.py

```diff
@@ -214,28 +212,25 @@ def delete_s3_acc_buckets_objects(self, s3_data: dict, obj_crud: bool = False,
                         response = s3_del.delete_multiple_objects(_bucket, obj_list[1], quiet=True)
                         LOGGER.debug("Delete multiple objects response %s", response)
                 return True, "Successfully performed Objects Delete operation"
-            if is_ver:
-                try:
-                    obj_name = v_etag.pop("obj_name")
-                    s3_ver = v_etag.pop("s3_ver")
-                except KeyError:
-                    return False, "Failed to get required data for deleting versioned objects"
-                for bucket in list(v_etag.keys()):
-                    LOGGER.info("Deleting object versions for %s", bucket)
-                    for v_e in v_etag[bucket]:
-                        v_id = list(v_e.keys())[0]
-                        s3_ver.delete_object_version(bucket=bucket, key=obj_name, version_id=v_id)
             for details in s3_data.values():
                 s3_del = S3TestLib(endpoint_url=S3_CFG["s3_url"],
-                                   access_key=details['accesskey'],
-                                   secret_key=details['secretkey'])
-                response = s3_del.delete_all_buckets()
-                if not response[0]:
-                    return response
+                                   access_key=details['accesskey'], secret_key=details['secretkey'])
+                s3_ver = S3VersioningTestLib(access_key=details['accesskey'],
+                                             secret_key=details['secretkey'],
+                                             endpoint_url=S3_CFG["s3_url"])
+                buckets = s3_del.bucket_list()[1]
+                LOGGER.info("Delete all versions and delete markers present for all the buckets.")
+                for _bucket in buckets:
+                    empty_versioned_bucket(s3_ver, _bucket)
```

Comment:
buckets will have all the buckets right versioned and unversioned?

Reply:
> buckets will have all the buckets right versioned and unversioned?

Yes

Reply:
What will happen in case of unversioned buckets?

Reply:
If there is no version ID associated with bucket, or it can't find any version ID, it will assign [] to a variable.
to_delete = list_response[1].get("Versions", []) and won't go for delete_object_version() call